frontend/javascripts/admin/dataset/dataset_add_zarr_view.tsx

```diff
@@ -0,0 +1,217 @@
+
+  function validateUrls(userInput: string) {
+    if (
+      (userInput.indexOf("https://") === 0 && userInput.indexOf("s3://") !== 0) ||
+      (userInput.indexOf("https://") !== 0 && userInput.indexOf("s3://") === 0)
+    ) {
+      throw new Error("Dataset URL must employ either the https:// or s3:// protocol.");
+    } else {
+      setSelectedProtocol(userInput.indexOf("https://") === 0 ? "https://" : "s3://");
+    }
+  }
+
+  async function handleExplore() {
+    if (datasourceUrl) {
+      let datasourceToMerge;
+      if (!usernameOrAccessKey || !passwordOrSecretKey) {
+        datasourceToMerge = await exploreRemoteDataset([datasourceUrl]);
+      } else {
+        datasourceToMerge = await exploreRemoteDataset([datasourceUrl], {
+          username: usernameOrAccessKey,
+          pass: passwordOrSecretKey,
+        });
+      }
+      if (datasourceToMerge) {
+        if (datasourceConfig) {
+          // TODO: check that both datasources have same voxel size else warning
+          let currentDatasource;
+          try {
+            currentDatasource = JSON.parse(datasourceConfig);
+          } catch (e) {
+            Toast.error("The loaded datasource config contains invalid JSON.");
+            return;
+          }
+          const layers = currentDatasource.dataLayers.concat(datasourceToMerge.dataLayers);
+          const uniqueLayers = _.uniqBy(layers, (layer: DataLayer) => layer.name);
+          currentDatasource.dataLayers = uniqueLayers;
+          currentDatasource.id.name = `merge_${currentDatasource.id.name}_${datasourceToMerge.id.name}`;
+          setDatasourceConfig(jsonStringify(currentDatasource));
+        } else {
+          setDatasourceConfig(jsonStringify(datasourceToMerge));
+        }
+      } else {
+        Toast.error("Exploring this remote dataset did not return a datasource.");
+      }
+    } else {
+      Toast.error("Please provide a valid URL for exploration.");
+    }
+  }
+
+  async function handleStoreDataset() {
+    if (datasourceConfig && activeUser) {
+      let configJSON;
+      try {
+        configJSON = JSON.parse(datasourceConfig);
+      } catch (e) {
+        Toast.error("The loaded datasource config contains invalid JSON.");
+        return;
+      }
+      const result = await storeRemoteDataset(
+        configJSON.id.name,
+        activeUser.organization,
+        datasourceConfig,
+      );
+      console.log(result);
+      if (result) {
+        onAdded(activeUser.organization, configJSON.id.name);
+      }
+    }
+  }
+
+  return (
+    // Using Forms here only to validate fields and for easy layout
+    <div style={{ padding: 5 }}>
+      <CardContainer title="Add Zarr Dataset">
+        Please enter a URL that points to the Zarr data you would like to import. If necessary,
+        specify the credentials for the dataset. More layers can be added to the loaded datasource
+        using the Add button. Once you have approved of the resulting datasource you can import it.
```

Comment:
```suggestion
        specify the credentials for the dataset. More layers can be added to the datasource specification below
        using the Add button. Once you have approved of the resulting datasource you can import it.
```

I think, I like that a bit better :)